ipq40xx: convert ZTE MF286D to DSA

Leo-PL opened the pull request:
Signed-off-by: Lech Perczak <lech.perczak@gmail.com>

Running the conversion from around the device introduction to master, albeit on older tree - just rebased now.
Port 5 is labeled as "LAN1/WAN" - keeping the intention of original author here, personally I'm running it as if it were "LAN1", the same as it its primary role on stock firmware.

robimarko commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-mf286d.dts:
```diff
@@ -87,6 +87,7 @@
 
 		ess-psgmii@98000 {
 			status = "okay";
+			/delete-node/ ethernet-phy@0;
```
Drop this, this node does not exist anymore at all.

Leo-PL replied:
Done. Rebased as well.

Leo-PL commented:
@CHKDSK88 please review ;-)

CHKDSK88 commented:
@Leo-PL 
I have one question: What about WAN MAC increment?
And one nitpick: I suggest keep alphabetical order in dts nodes.

Leo-PL commented:
@robimarko is it possible to assign MAC to particular switch port by nvmem-cells? Or I need to fall-back to setting it in `ipq40xx_setup_macs` by hand?
@CHKDSK88 was the MAC setup exactly the same as in stock firmware? I don't have any unit with stock FW left, and maybe you do remember. In MF286{,A,R} it was a mess.

In the meantime, rebased and fixed node ordering.

robimarko commented:
Yes, you add nvmem cells to each of the switch ports to set the MAC, that is the preferred way

Leo-PL commented:
Ok then, added WAN MAC through nvmem-cells, and it works very well indeed.

CHKDSK88 commented:
> @CHKDSK88 was the MAC setup exactly the same as in stock firmware? I don't have any unit with stock FW left, and maybe you do remember. In MF286{,A,R} it was a mess.

@Leo-PL 
MF286D have only one MAC per device. Upstream accepted separate addresses for LAN, WAN and both Wifi. 

Now  looks  great. I will try it next week.

Leo-PL commented:
OK. I was wondering, that since we have the modem part working perfectly now, should we keep the WAN/LAN split, since this isn't the most typical setup for this device. In my personal builds, I include this port in LAN part, just as in a default configuration - mainly because on swconfig, ports assigned to gmac1 had broken VLANs.

robimarko commented:
Can you just let me know when its ready to be picked?

Leo-PL commented:
@CHKDSK88 kind reminder :-)

CHKDSK88 commented:
@Leo-PL 
sorry for mega lag. ;)

Everything work as expected.
`Tested-by: Pawel Dembicki <paweldembicki@gmail.com>`

Leo-PL commented:
@robimarko ping ^

robimarko commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-mf286d.dts:
```diff
@@ -319,6 +325,33 @@
+
+&swport5 {
+	status = "okay";
+	label = "wan";
```
This is already the default label for swport5.
I will drop it during picking

robimarko commented:
Picked, thanks
